Mathlib/RingTheory/AlgebraicIndependent.lean

```diff
@@ -741,3 +743,68 @@ theorem algebraicIndependent_empty [Nontrivial A] :
   algebraicIndependent_empty_type
 
 end Field
+
+section RankAndCardinality
+
+open Cardinal
+
+theorem IsTranscendenceBasis.lift_cardinalMk_eq_max_lift
+    {F : Type u} {E : Type v} [CommRing F] [Nontrivial F] [CommRing E] [IsDomain E] [Algebra F E]
+    {ι : Type w} {x : ι → E} [Nonempty ι] (hx : IsTranscendenceBasis F x) :
+    lift.{max u w} #E = lift.{max v w} #F ⊔ lift.{max u v} #ι ⊔ ℵ₀ := by
+  let K := Algebra.adjoin F (Set.range x)
+  suffices #E = #K by
+    rw [this, ← lift_mk_eq'.2 ⟨hx.1.aevalEquiv.toEquiv⟩]
+    simp
+  haveI : Algebra.IsAlgebraic K E := hx.isAlgebraic
+  refine le_antisymm ?_ (mk_le_of_injective Subtype.val_injective)
+  obtain ⟨i⟩ := ‹Nonempty ι›
+  haveI : Infinite K := hx.1.aevalEquiv.toEquiv.infinite_iff.1 inferInstance
+  simpa only [sup_eq_left.2 (aleph0_le_mk K)] using Algebra.IsAlgebraic.cardinalMk_le_max K E
+
+theorem IsTranscendenceBasis.lift_rank_eq_max_lift
+    {F : Type u} {E : Type v} [Field F] [Field E] [Algebra F E]
+    {ι : Type w} {x : ι → E} [Nonempty ι] (hx : IsTranscendenceBasis F x) :
+    lift.{max u w} (Module.rank F E) = lift.{max v w} #F ⊔ lift.{max u v} #ι ⊔ ℵ₀ := by
+  let K := IntermediateField.adjoin F (Set.range x)
+  haveI : Algebra.IsAlgebraic K E := hx.isAlgebraic_field
+  rw [← rank_mul_rank F K E, lift_mul, ← hx.1.aevalEquivField.toLinearEquiv.lift_rank_eq,
+    MvRatFunc.rank_eq_max_lift, lift_max, lift_max, lift_lift, lift_lift,
+    lift_aleph0]
+  refine mul_eq_left le_sup_right ((lift_le.2 ((rank_le_card K E).trans
+    (Algebra.IsAlgebraic.cardinalMk_le_max K E))).trans_eq ?_) (by simp [rank_pos.ne'])
+  rw [lift_max, ← lift_mk_eq'.2 ⟨hx.1.aevalEquivField.toEquiv⟩]
+  simp
```

Comment:
```suggestion
    MvRatFunc.rank_eq_max_lift, lift_max, lift_max, lift_lift, lift_lift, lift_aleph0]
  refine mul_eq_left le_sup_right ((lift_le.2 ((rank_le_card K E).trans
    (Algebra.IsAlgebraic.cardinalMk_le_max K E))).trans_eq ?_) (by simp [rank_pos.ne'])
  simp [lift_max, ← lift_mk_eq'.2 ⟨hx.1.aevalEquivField.toEquiv⟩]
```

Reply:
`lift_max` is a simp lemma, but I need to check if it works if it's removed from your last line of `simp`.